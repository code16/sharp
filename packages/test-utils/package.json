{
    "name": "@sharp/test-utils",
    "private": true,
    "main": "src/index.js",
    "version": "1.0.0"
}