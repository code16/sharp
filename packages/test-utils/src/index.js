export { default as MockInjections } from './MockInjections';
export { default as MockI18n } from './MockI18n';
export { default as QueryComponent } from './QueryComponent';
export { default as MockTransitions } from './MockTransitions';
export { mockSFC, unmockSFC } from './mockSFC';
export { mockProperty, unmockProperty, setter } from './mock-utils';
export { nextRequestFulfilled } from './moxios-utils';

export function wait(delay) {
    return new Promise(resolve => {
        setTimeout(resolve, delay);
    });
}