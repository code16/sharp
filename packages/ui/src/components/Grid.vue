<template>
    <div class="SharpGrid">
        <template v-for="row in rows">
            <div class="SharpGrid__row row" :class="rowClass(row)">
                <template v-for="col in row">
                    <div class="SharpGrid__col" :class="colClasses(col)" v-empty-class="'SharpGrid__col--empty'">
                        <slot :item-layout="col" />
                    </div>
                </template>
            </div>
        </template>
    </div>
</template>

<script>
    import { emptyClass } from 'sharp/directives';

    export default {
        name: 'SharpGrid',

        props: {
            rows: { // 2D array [row][col]
                type: Array,
                required: true
            },
            rowClass: {
                type: Function,
                default: () => null,
            },
            colClass: {
                type: Function,
                default: () => null,
            },
        },
        methods: {
            colClasses(col) {
                const { size, sizeXS } = col;
                const hasSize = !!size;
                return [
                    {
                        [`col-${sizeXS}`]: sizeXS,
                        [`col-md-${size}`]: hasSize,
                        'col-md': !hasSize
                    },
                    this.colClass(col),
                ];
            }
        },
        directives: {
            'empty-class': emptyClass
        }
    }
</script>
