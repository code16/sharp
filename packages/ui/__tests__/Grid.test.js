import Grid from '../src/components/Grid.vue';
import { mount } from '@vue/test-utils';

describe('grid',()=>{
    let wrapper, baseOptions;
    beforeEach(()=>{
        wrapper = mount(Grid, baseOptions={
            propsData: {
                rows: [
                    [{ size: 6, sizeXS: 12 }, { size: 4 }],
                    [{ size: 10 }, { size: null }] // should have class col-sm
                ]
            }
        });
    });

    test('can mount Grid', ()=>{
        expect(wrapper.html()).toMatchSnapshot();
    })

});