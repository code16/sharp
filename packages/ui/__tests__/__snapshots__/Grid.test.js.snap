// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`grid can mount Grid 1`] = `
<div class="SharpGrid">
	<div class="SharpGrid__row row">
		<div class="SharpGrid__col col-12 col-md-6 SharpGrid__col--empty"></div>
		<div class="SharpGrid__col col-md-4 SharpGrid__col--empty"></div>
	</div>
	<div class="SharpGrid__row row">
		<div class="SharpGrid__col col-md-10 SharpGrid__col--empty"></div>
		<div class="SharpGrid__col col-md SharpGrid__col--empty"></div>
	</div>
</div>
`;
